cpp/src/fil/fil.cu

```diff
@@ -586,20 +586,35 @@ void tl2fil_common(forest_params_t* params, const tl::Model& model,
   // fill in forest-dependent params
   params->depth = max_depth(model);  // also checks for cycles
 
+  const tl::ModelParam& param = model.param;
+
   // assuming either all leaves use the .leaf_vector() or all leaves use .leaf_value()
   size_t leaf_vec_size = tl_leaf_vector_size(model);
   if (leaf_vec_size > 0) {
     ASSERT(leaf_vec_size == model.num_output_group,
            "treelite model inconsistent");
     params->num_classes = leaf_vec_size;
     params->leaf_payload_type = leaf_value_t::INT_CLASS_LABEL;
+    ASSERT(
+      param.pred_transform != "identity_multiclass",
+      "multiclass predict() only supported by choosing most likely label,\n"
+      "please use pred_transform == 'max_index' ('sigmoid' is still supported,"
+      " with 'max_index' implied)");
+    if ((std::string(param.pred_transform) == "max_index") &&
+        !(tl_params->output_class && tl_params->threshold == 0.5)) {
+      ASSERT(false,
+             "pred_transform == 'max_index' needs output_class && "
+             "threshold == 0.5 to be faithfully executed. Otherwise, please "
+             "use 'identity'");
+      // 'max_index' will be equivalent to setting 'output_class' and threshold == 0.5
+    }
   } else {
     params->leaf_payload_type = leaf_value_t::FLOAT_SCALAR;
```

Comment:
This branch should check for `pred_transform` values that are supported for binary classification and regression.

Reply:
The if else statement here is to check if the model is a classification model or a regression model. If it is a regression model then leaf_vec_size=0 and it executes the statements in else. I think a new if statement with condition `leaf_vec_size <=2` might help to reduce duplication of the code.

Also which are the options that can only be used for binary classification/ regression?  The fil.h file and treelites pred_transform.cc file (https://github.com/dmlc/treelite/blob/7f01e631da8687189473ad6b177ba0615b19496b/src/compiler/pred_transform.cc#L44) state that `sigmoid` can only be used for binary classification/ regression. However, the original fil_test.cu uses `sigmoid` for multi-class classification problems as well.

Reply:
It's be better to have a sequence of asserts, each checking one condition only, e.g.:

```
ASSERT(tl_params->output_class, "output_class==true is required for multi-class models");

std::string pred_transform(param.pred_transform);
ASSERT(pred_transform == "max_index" || pred_transform == "identity_multiclass", 
"only max_index and identity_multiclass values of pred_transform "
"are supported for multi-class models");
```

Regarding FIL tests that use sigmoid for multi-class: feel free to remove them. It's best to do it in another pull request, but feel free to do it in this pull request if this change causes them to fail.